### DISCUSSION TITLE
v1.10.0 Release - early access build

### DISCUSSION THREAD

**Discussion by jjw24** [Flow-Launcher/Flow.Launcher#956] [Announcements]
This is the future v1.10.0 release, you can test it by using the [portable build](https://github.com/Flow-Launcher/Prereleases/releases/latest/download/Flow-Launcher-Portable.zip).

If you would like to use the installed version instead, here's the link. [Flow-Launcher-Setup.exe](https://github.com/Flow-Launcher/Prereleases/releases/latest/download/Flow-Launcher-Setup.exe)

We recommend when testing this, to use the portable version and keep it separate from your everyday version because there may be changes to some of the setting files which may not be backwards compatible with the current release. If you are sure you will be using this release going forward, then go ahead and replace its UserData folder with your current one's or download the installed version. 

Version of this early access build is the **same as the version of the current latest release**, this is intended.

Any issues, let us know below.

**Comment by raindropsfromsky** [comment]
In the **web searches** plugin, when we add a new website, the instruction reads as follows:

If you have a web search service you want to use, you can add it to Flow. For
example, you can follow the url format in the address bar if you want to search
casino on Netflix: "https://www.netflix.com/search?q=Casino". To do this,
change the search term 'Casino' as follows.
https://www.netflix.com/search?q={q}
Add it to the URL section below. You can now search Netflix with Flow using any
search terms.

This is a bit confusing.

A better version is as follows:

If you want to add a search for a particular website to Flow, first enter a dummy text string in the search bar of that website, and launch the search. Now copy the contents of the browser's address bar, and paste it in the **URL** field below. Replace your test string with **{q}**.

For example, if you search for **casino** on Netflix, its address bar reads **https://www.netflix.com/search?q=Casino**. 
Now copy this entrire string and paste it in the **URL** field below.
Then replace **casino** with **{q}**. 
Thus, the generic formula for a search on Netflix is **https://www.netflix.com/search?q={q}**

**Reply by jjw24** [reply]
I think your version is clearer and we can update it. I will make an issue for it.

**Reply by jjw24** [reply]
https://github.com/Flow-Launcher/Flow.Launcher/issues/1309 created.

**Comment by raindropsfromsky** [comment]
This is not exactly a bug, but the **web searches** plugin is missing in the [plugin directory](https://www.flowlauncher.com/docs/#/plugins).

**Reply by taooceros** [reply]
It is embedded in Flow so don't need to present there.

**Reply by raindropsfromsky** [reply]
But it is listed along with the other plugins!

If this is an integral part of FL, it should not be listed as a plugin, right?

**Reply by taooceros** [reply]
> But it is listed along with the other plugins!
> 
> If this is an integral part of FL, it should not be listed as a plugin, right?

Well everything is presented in plugin form theoretically to have a uniform interface. We just provide some built in Plugins that we think user may want.

**Reply by jjw24** [reply]
This is fixed with the early access build above now :)

**Comment by raindropsfromsky** [comment]
In the **Settings**, the **Plugin Store** tab shows a large list of plugins.

This screen should show which of these plugins are already installed.

Also, there should be a filter to show only those filters that are installed (or not installed).
- The "show only installed plugins" filter will allow us to check whether to disable any of them.
- The "show only not-installed plugins" filter will allow us to focus on new plugins

**Reply by jjw24** [reply]
Good suggestion, I thought we already have an issue for this but I can't find one so will make this into an issue.

**Reply by jjw24** [reply]
https://github.com/Flow-Launcher/Flow.Launcher/issues/1310 created

**Reply by deefrawley** [reply]
@jjw24 Just loaded the pre-release today. Installed my currency plugin and the windows terminal plugin. In the Settings -> Plugin Store only the windows terminal plugin is showing as installed (however in Settings - Plugin I can see the currency plugin and I have no issues using it).

**Comment by raindropsfromsky** [comment]
Bugs and suggestions for **Themes** tab in **Settings**:
1. When we toggle the **Query window shadow effect** switch, its effect cannot be seen in the preview window, because it has a black background, because of which we cannot see the shadow. 
2. when we toggle the **Query window shadow effect** switch, the query window shifts a bit in the preview! <br>Why should it move?
3. Flow Launcher has a nice feature of showing the matching letters in a different color (e.g. blue). <br>But the query window does not show how different themes affect this highlighted matching text. <br> For example, can it change the color of the highlighted text?
4. Please provide a separate control to change the color of the text that matches the query string.
5. Before coming to the **Themes** tab, the app already has a default theme. <br> The Themes tab allows us to select another theme. <br>But there is no control to revert to the original theme!<br>**Suggestion:** Please provide a "Default" button, which should be selected by default in the beginning. This button should be set apart from the other themes, so the user knows how to leave other themes and revert to the default look. 
6. The preview window should represent the screen of the laptop, and show the size of the query box accordingly. <br>My laptop has FHD monitor, but if I set the query box width to 720 pixels, the preview shows that it occupies the full width of the screen!
7. The preview window should be resizable till the full width of the screen. <br>But although I have an FHD monitor, the window can be resized only up to 900 pixels. It should have been resizable up to 1920 pixels.

**Reply by raindropsfromsky** [reply]
There is no response to this post. Should I repeat these points in the **Issues** section?

**Reply by jjw24** [reply]
Hey how's it going?

These are good points, I just haven't got around to investigating these bugs yet, let's leave them here and when we are ready to start them we can break them into individual issues.

**Reply by raindropsfromsky** [reply]
oh, sure! 

Thanks for the prompt response!

**Comment by raindropsfromsky** [comment]
The **magnifying glass** icon in the query box has no practical use. It ends up wasting precious real estate.
Instead, it should indicate when Flow Launcher is scanning the disks (e.g. when we attach HDDs or pen drives to the laptop).
(For example, the Listary icon rotates while it is updating its indexes.) 

Also, when I have downloaded a few files, they do not appear in the Flow Launcher's search results for a few seconds.
This delay shows that FL cannot update its indexing in real-time.
If this is by design, then the icon should reflect when the indexes are out of sync.

**Reply by raindropsfromsky** [reply]
There is no response to this post. Should I repeat these points in the **Issues** section?

**Reply by jjw24** [reply]
I agree, the magnifying glass can be used to indicate these loading progress. Once we start work on https://github.com/Flow-Launcher/Flow.Launcher/issues/55, we will be able to extend it to your suggestions.

 >Also, when I have downloaded a few files, they do not appear in the Flow Launcher's search results for a few seconds.
This delay shows that FL cannot update its indexing in real-time.
If this is by design, then the icon should reflect when the indexes are out of sync.

I believe when you do this it's with Explorer/Everything plugin, which the indexing is provided by Windows Search/Everything and we would probably not be able to get a response back from these services on their indexing progress. I do like the idea though.

**Reply by jjw24** [reply]
>There is no response to this post. Should I repeat these points in the Issues section?

Can leave this suggestion here for now. I will take another look at them when we start work on the magnifying glass change.

**Reply by raindropsfromsky** [reply]
ok, thanks!

**Comment by raindropsfromsky** [comment]
When I hover my mouse on the search results, a tooltip appears. 
IMHO it is not wide enough, and unnecessarily folds the text in two lines.
Why should I have to read a file name in two lines? It is not comfortable!

There are no controls to customize this tooltip.
For example, I should be able to change its width, font type+size+color, background color and toggle its shadow.

**Reply by raindropsfromsky** [reply]
I saw an animation on the search box border (a blue accent was running left to right for some time).
There is no documentation on that.
Does that indicate that FL is scanning the disks to update its indexes?
Or does that signify something else?

**Reply by taooceros** [reply]
The tool tip is a ready to use control, and I am not sure whether we can adjust that. We will take a look later.

**Reply by taooceros** [reply]
> I saw an animation on the search box border (a blue accent was running left to right for some time).
> There is no documentation on that.
> Does that indicate that FL is scanning the disks to update its indexes?
> Or does that signify something else?

It is loading plugin results. Some Plugins takes longer to load result (not necessarily updating index).

**Comment by raindropsfromsky** [comment]
On the **plugin Stores** page, there should be a search facility, so that we can quickly locate a plugin for the desired functionality.

**Reply by taooceros** [reply]
Working on it 😊

**Comment by raindropsfromsky** [comment]
Hi

AFAICT the last development version (1.10.0) was released six months ago. This is too long for a development cycle. Are there any later versions released for us to test? 

How would we know when a version is released?

Thanks in advance!

**Reply by taooceros** [reply]
Actually 1.10 is a nightly build, so it evolve over time. You can check the commit history of what 1.10 has been improved. The latest released is 1.9.4, which only involved some technical fixes.

I agree with you that our development cycle is way too long, but because most of us work on flow in our free time, it is hard to have a consistent time of developing new features and reviewing code. I would like to have a faster pace of development but it seems it's tough for now.

**Reply by taooceros** [reply]
If a new version is released, we will add that to the release page and make an announcement.

**Reply by jjw24** [reply]
> The latest released is 1.9.4, which only involved some technical fixes.

1.9.3 actually, 1.9.4 will be released in the immediate future with 1.10.0 on the horizon. 

@raindropsfromsky the artifact here is the nightly build of all the latest stable features so if you are after the latest and greatest you can download it here.

1.10.0 is still not ready because there are still a couple of very nice improvements we have yet had the time to review/finish off.

**Comment by yuwen-ye** [comment]
The link of [portable build](https://ci.appveyor.com/api/projects/JohnTheGr8/flow-launcher/artifacts/Output%2FPackages%2FFlow-Launcher-Portable.zip?branch=master&pr=true) given in the OP redirects to: 
https://ci.appveyor.com/api/projects/JohnTheGr8/flow-launcher/artifacts/Output%2FPackages%2FFlow-Launcher-Portable.zip?branch=master&pr=true
When I unzipped the file downloaded, it turned out the build was 1.9.4 (current stable version). 

I assume the link actually should be **?branch=dev**, is that correct?

**Reply by taooceros** [reply]
not really. This actually means the branch that is going to be merged into `master`, and only `dev` is the branch that is going to be merged into master. It is somewhat confusing though.

**Reply by yuwen-ye** [reply]
However when I unzipped the file downloaded, it turned out the build is 1.9.4 (current stable version). 

Is there something wrong with my steps?

**Reply by taooceros** [reply]
No it is just we haven't updated the version number.
